internal/controller/spireserver_controller_mock_test.go

```diff
@@ -107,6 +107,16 @@ func TestSpireserverController(t *testing.T) {
 	}
 }
 
+func TestCheckTrustDomain(t *testing.T) {
```

Comment:
Given that more test cases will be added for `checkTrustDomain`, I would recommend renaming this to clarify which test case is being checked (e.g., valid DNS name, invalid DNS name, string of chars, string of nums, etc.).

Reply:
Sounds good. I will look into this.